activesupport/lib/active_support/cache/serializer_adapter.rb

```diff
@@ -0,0 +1,171 @@
+# frozen_string_literal: true
+
+require_relative "entry"
+
+module ActiveSupport
+  module Cache
+    class SerializerAdapter # :nodoc:
```

Comment:
`SerializerAdapter` is a coder with swappable serializer and compressor.

Is the primary concern integration with something like paquito?

What if we (1) rename `SerializerAdapter` to `Cache::Coder`, and (2) support specifying either a `:coder` option _OR_ `:serializer` and `:compressor` options for `Cache::Store`?  When `:coder` is specified, it will be used directly, no `Cache::Coder` involved.  Otherwise, `:serializer` and `:compressor` will be used via `Cache::Coder`, with their default values controlled by the cache format version.

Reply:
> Is the primary concern integration with something like paquito?

In part yes. I think if a `coder` is specified, it should have total control over the serialization, Active Support shouldn't prepend anything to the provided payload etc.

>  support specifying either a :coder option OR :serializer and :compressor options

I think you touch on something here.

Paquito went the route of taking total control over how the `Entry` (and its content) is serialized.

You can't combine that with a configurable compressor. But in many case users may only want to control how the `Entry#value` is serialized.

If both were explicitly distinct I think it would make the code clearer.

So 👍 to experiment that (and sorry for dragging my feet on this, I know you've put a lot of efforts into it, I'm just super worried about the maintainability of all this).